### ISSUE TITLE
PDF generated by OnlyOffice is extremely large when third-party CJK fonts are used

### ISSUE THREAD

**Issue by mon-jai** [ONLYOFFICE/DocumentServer#2934]
PDFs generated by OnlyOffice are extremely large when third-party CJK fonts are used.

Please see the following demo document containing two lines of Chinese text: 
[Demo Document.docx](https://github.com/user-attachments/files/17478607/Demo.Document.docx)

The PDF generated by OnlyOffice is a massive 13MB: 
[Demo Document.pdf](https://github.com/user-attachments/files/17478608/Demo.Document.pdf)

However, the same PDF is only 15.25KB when opened in LibreOffice Draw and re-exported: 
[Demo Document - Libreoffice.pdf](https://github.com/user-attachments/files/17478609/Demo.Document.-.Libreoffice.pdf)

The font used in the file is Source Han Sans TC v2.004 (https://github.com/adobe-fonts/source-han-sans/releases/download/2.004R/SourceHanSansHWTC.zip).

**Comment by TatianaKirillova** [comment]
Hello @mon-jai 
Thank you for contacting us! This issue has already been discussed in the post #2739

**Comment by mon-jai** [comment]
@TatianaKirillova It doesn't seem like a duplication.

In my demo document, there are no form fields—just two lines of text. Yet, the entire font is included.

**Comment by TatianaKirillova** [comment]
@mon-jai Yes, but you can always add them - at any time. Therefore, we are forced to save the font. If you do not plan to add form fields, then you can simply save docx->pdf

**Comment by mon-jai** [comment]
@TatianaKirillova 

> Yes, but you can always add them - at any time.

I consider a PDF without form fields more like a non-editable format. In most cases, no one will attempt to edit it.

> then you can simply save docx->pdf

Yeah, I just did that (either by 'Save As' or 'Print as PDF'), and in both cases, the resulting PDF is 13MB.

There is no known way to create a smaller PDF in OnlyOffice with only the used glyphs embedded.

**Comment by mon-jai** [comment]
@TatianaKirillova 

> Yes, but you can always add them - at any time.

I found a PDF without form fields to be more like a non-editable format. In most cases, no one will attempt to edit it.

> then you can simply save docx->pdf

Yeah, I just did that (either by "Save As"  or "Print as PDF"), and in both cases, the resulting PDF is 13MB.

There is no known way to export a smaller PDF in OnlyOffice with only the used glyphs embedded (that's the whole point).

**Comment by TatianaKirillova** [comment]
@mon-jai I save your document in pdf format in the ONLYOFFICE editor and it weighs 49 KB
[Demo.Document (1).pdf](https://github.com/user-attachments/files/17510065/Demo.Document.1.pdf)

**Comment by mon-jai** [comment]
> The font used in the file is Source Han Sans TC v2.004 (https://github.com/adobe-fonts/source-han-sans/releases/download/2.004R/SourceHanSansHWTC.zip).

@TatianaKirillova Can you try again with [Source Han Sans TC v2.004](https://github.com/adobe-fonts/source-han-sans/releases/download/2.004R/SourceHanSansHWTC.zip) installed?

This issue only occurs with third-party CJK fonts. Since you don’t have that font installed, OnlyOffice falls back to the system font, which may not be embedded in the generated PDF.

**Comment by TatianaKirillova** [comment]
@mon-jai Yes, you're right. We try to save the file with text, not pictures (like other editors), which results in increased weight. I created a proposal # 71235 so that it would be possible to choose in which format to save the file - lighter weight or it is important that it be saved as text.

**Comment by mon-jai** [comment]
Thank you! Could you reopen the issue?

Also, what happens if a user chooses the "lighter weight" option once it’s implemented? Will the font be subsetted?

**Comment by mon-jai** [comment]
@TatianaKirillova Thank you! Could you reopen the issue?

Also, what happens if a user chooses the "lighter weight" option once it’s implemented? Will the font be subsetted?

**Comment by TatianaKirillova** [comment]
Hello @mon-jai 
I'm happy to report that the issue is fixed in [this commit](https://github.com/ONLYOFFICE/core/commit/fa9e3cb48efaa253014b1c54ac8a17802178a4c3) and will be included in the hotfix. Don't miss updates. Have a good day!

**Comment by Rita-Bubnova** [comment]
DocumentServer v9.1.0 is released so I close this issue. Feel free to comment or reopen it if you got further questions.
